### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by devmotion**
I updated CI a bit, including the invalidation action. The last commit is a generalization that should handle differently named default branches as well, not sure if it should be applied to the other actions as well.

### CODE REVIEW

**Comment by devmotion** [comment]
Haha seems the generalization is not completely correct :smile:

**Comment by devmotion** [comment]
Seems to work now.

**Comment by ChrisRackauckas** [comment]
@devmotion could you run the mass action installer with this? Thanks.

**Comment by ChrisRackauckas** [comment]
@devmotion could you run the mass action installer with this? Thanks.

File: .github/workflows/Invalidations.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Invalidations
+
+on:
+  pull_request:
```
Comment (ranocha) [inline .github/workflows/Invalidations.yml:4]: @devmotion: Didn't you want to restrict it to PRs to `master`?

**Comment by ranocha** [comment]
I made PRs to the DiffEq repos where I have access - they are few enough to not bother with MassInstallAction.jl

**Comment by ranocha** [comment]
Someone else needs to do it for the remaining repos with MassInstallAction.jl (or I would need more rights in SciML).

**Comment by devmotion** [comment]
Was sleeping but ran MassInstallAction now. Seems to work, only failing repositories were the archived ones. The only problem is that I hit Github's rate limit now and hence starting from SciMLNLSolve I was not allowed to open any PRs anymore. But the branches exist, so one just has to open the PRs manually (or write a script for it).

**Comment by devmotion** [comment]
You could have saved yourself some time if you would have waited for the MassInstallAction run @ranocha :slightly_smiling_face:

**Pull request merged**